scripts/pre-commit.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# It is recommended to execute the script before commit
+# which will help us to reduce test/fmt/clippy failures in CI
+
+start=$(date +%s)
+
+make fmt
+taplo fmt
+
+root_dir=$(git rev-parse --show-toplevel)
+CARGO_TARGET_DIR=${root_dir}/tmp
```

Comment:
> Looks good overall, thanks. I have left comments for several small things.
> 
> A general question though: instead of inserting the `status` to the opaque vec, what prevents us from simply dropping the failed extrinsics when importing paentchain blocks? (handling in `parentchain_handler.rs`)

Dropping failed extrinsics might not be a good solution. In certain cases, we may need to know which extrinsics failed or need to count which extrinsics failed. What do you think?

Reply:
Actually, this script just is a patch to the original pre-commit (clippy, unit test)